src/figuresClasses.ts

```diff
@@ -1,19 +1,95 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape,
+  color: Color,
 
+  getArea: () => number,
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  readonly shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (
+      a >= b + c
+      || b >= a + c
+      || c >= a + b
+      || a <= 0
+      || b <= 0
+      || c <= 0
+    ) {
+      throw new Error('Impossible triangle');
+    }
+
+    this.a = a;
+    this.b = b;
+    this.c = c;
```

Comment:
You don't need these lines of code,  the properties were added when you wrote them as constructor parameters with access modifiers.